Solution for broken SMP DQM codes for validations

sasharma6 opened the pull request:
" Poblem : There were missing EwkElecDQM & EwkMuonDQM modules in the physics directory of samples ZEE, ZMM, WE WM. 
Which leads to no validation of CMSSW releases for 81X and few 80X.
Solution implemented : As suggested by the group people we have removed the trigger part from both EwkElec and EwkMuon DQM codes and using CMSSW 810pre9 we obtained the plots.
Things to do: To approve the code and check whether it is properly merged"

sasharma6 committed:
indented code

cmsbuild commented:
A new Pull Request was created by  @sasharma6  for CMSSW_8_1_X.

It involves the following packages:

DQM/Physics


@cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@rappoccio this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

dmitrijus commented:
Please set the proper title and the detailed description.
For example, such as in this pull request: https://github.com/cms-sw/cmssw/pull/15830

sasharma6 commented:
Hi Dmitrijus,

I have added the information for PR#16322. Kindly have a look.

Thanks and regards
Sandeep

Get Outlook for Android<https://aka.ms/ghei36>



On Fri, Oct 28, 2016 at 7:18 PM +0530, "Dmitrijus" <notifications@github.com<mailto:notifications@github.com>> wrote:


Please set the proper title and the detailed description.
For example, such as in this pull request: #15830<https://github.com/cms-sw/cmssw/pull/15830>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/16322#issuecomment-256924359>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ATMpCvxwiCUt1RqJiBUvN_LLKajMrabGks5q4f0IgaJpZM4KecS6>.

dmitrijus commented on DQM/Physics/src/EwkElecDQM.cc:
```diff
@@ -1,1039 +1,955 @@
+{
+    double all = nall;
+    double esel = nsel / all;
+    LogVerbatim("") << "\n>>>>>> SELECTION SUMMARY BEGIN >>>>>>>>>>>>>>>";
```
Why does this use LogVerbatim? Why not LogWarning?

dmitrijus commented on DQM/Physics/src/EwkElecDQM.cc:
```diff
@@ -1,1039 +1,955 @@
+    // Beam spot
+    Handle < reco::BeamSpot > beamSpotHandle;
+    if (!iEvent.getByToken(beamSpotTag_, beamSpotHandle)) {
+        std::cout << ">>> No beam spot found !!!";
```
No std::cout or std::cerr!

dmitrijus commented on DQM/Physics/src/EwkElecDQM.cc:
```diff
@@ -1,1039 +1,955 @@
+        int out2 = 0;
+        if (out2 == 1) {
+            if (isBarrel) {
+                std::cout << "\t... detain value " << detain << " (barrel), pass? " << electron_sel[3];
```
std::cout used again!

dmitrijus commented on DQM/Physics/src/EwkMuDQM.cc:
```diff
@@ -1,878 +1,967 @@
+    using namespace edm;
+    Handle < View < Muon > >muonCollection;
+    if (!iEvent.getByToken(muonTag_, muonCollection)) {
+        std::cout << ">>> Muon collection does not exist !!!" << std::endl;
```
!

dmitrijus commented on DQM/Physics/src/EwkMuDQM.cc:
```diff
@@ -1,878 +1,967 @@
+    unsigned int muonCollectionSize = muonCollection->size();
+    Handle < reco::BeamSpot > beamSpotHandle;
+    if (!iEvent.getByToken(beamSpotTag_, beamSpotHandle)) {
+        std::cout << ">>> No beam spot found !!!" << std::endl;
```
!

dmitrijus reviewed (changes requested):
Looks much better than previous pull requests.
One major issue is that thread-unsafe std::cout is used. It should be converted into LogWarning (where appropriate). 

Please make a PR to 90x with std::cout removed.
Once it is *merged*, you will be able to submit backports to other branches. However, it is much easier to keep 1 PR for all the discussions.

In general, you should edit the PR instead of making a new for every single change (there is an 'edit' button).

sasharma6 committed:
Update EwkElecDQM.cc

 std::cout  which is unsafe thread has been replaced by LogWarning in " No beam spot found" , Line no 362, and replaced by LogTrace in " "\t... detain value  "\ line no  594 to 597. No change LogVerbatim as it is start of selection summary , end of selection summary is also by LogVerbatim

cmsbuild commented:
Pull request #16322 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

sasharma6 committed:
Update EwkMuDQM.cc

std::cout has been replaced by LogWarning("")

cmsbuild commented:
Pull request #16322 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

dmitrijus commented:
Please close this PR. And the other one in 81x.
Your main target is the upstream branch - 90x - https://github.com/cms-sw/cmssw/pull/16630
The PR there has a merge conflict, I cannot start the tests. Fix it and I will start the tests.
